packages/firestore/src/local/tab_notification_channel.ts

```diff
@@ -102,6 +171,81 @@ export class LocalStorageNotificationChannel implements TabNotificationChannel {
     this.persistState();
   }
 
+  addMutation(batchId: BatchId): void {
+    this.pendingBatchIds[batchId] = this.instanceId;
```

Comment:
I assume the intention is that these write to local storage? Again, more comments and less code would be more helpful.